scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2020 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.animate_images_with_image_overlay;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.layers.ArcGISTiledLayer;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Surface;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Camera;
+import com.esri.arcgisruntime.mapping.view.ImageFrame;
+import com.esri.arcgisruntime.mapping.view.ImageOverlay;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Slider;
+
+public class AnimateImagesWithImageOverlayController {
+
+  @FXML
+  private SceneView sceneView;
+  @FXML
+  private Button controlAnimationButton;
+  @FXML
+  private Slider opacitySlider;
+  @FXML
+  private ComboBox<String> framesComboBox;
+
+  private List<ImageFrame> imageFrames;
+
+  private Integer imageIndex = 0;
+  private Integer period = 67;
+
+  private Timer timer = null;
+  private boolean isTimerRunning = true;
+
+  public void initialize() {
+
+    try {
+
+      // create a new ArcGISScene and set it to the scene view
+      ArcGISScene scene = new ArcGISScene();
+      sceneView.setArcGISScene(scene);
+
+      // create a camera, looking at the pacific southwest sector
+      Point observationPoint = new Point(-116.621, 24.7773, 856977.0);
+      Camera camera = new Camera(observationPoint, 353.994, 48.5495, 0.0);
+
+      // create an envelope of the pacific southwest sector for displaying the image frame
+      Point pointForImageFrame = new Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84());
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
```

Comment:
Can we just define an envelope straight away with four corner points, instead of `Evelope(center, width, height)`?

Reply:
so the current implementation ensures the image frame and viewpoint sits exactly in the correct geographic location for the data provided. I could change it to an envelope with four corners if you thought there was a significant benefit in doing so over the existing implementation, though it ends up in the same lines of code plus a change in setting up the viewpoint logic. As always there's more than one way to skin a cat! I'm just not sure of the benefit of the 4 point envelope over the existing implementation.

Reply:
I just find the constructur `Envelope(center, width, height)` incredibly unintuitive. We aren't extracting the geographic location from the images or any other data source, so there's not really much of a point in doing it that way.The way we are creating the envelope up there is essentially the same as adding/subtracting (-120.0724273439448, 35.131016955536694) from the x/y of the centre-point, respectively. We could just write these values out so it's clearer where in the world these corner points are. But yeah, a matter of preference!